Source/JavaScriptCore/API/glib/JSCValue.cpp

```diff
@@ -2145,3 +2145,43 @@ char* jsc_value_to_json(JSCValue* value, unsigned indent)
 
     return json;
 }
+
+/**
+ * jsc_value_new_promise:
+ * @context: a #JSCContext
+ * @executor: (scope call) (nullable) (closure user_data): an initialization callback
+ * @user_data: (nullable): user data passed in @executor
+ *
+ * Create a new Promise returning handles to resolve and reject method
```

Comment:
This is not accurate now, we should document here that executor will be invoked during promise initialization and it receives the resolve and reject objects than can be called to resolve or reject the promise. We should also document that the executor is called like a js function, so exceptions raised during the executor invocation will  not be propagated to the context, but handled by the promise causing a rejection.